example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java

```diff
@@ -0,0 +1,147 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.sampleapp;
+
+import static java.lang.Double.max;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+import androidx.car.app.CarContext;
+import androidx.car.app.model.Action;
+import androidx.car.app.model.ActionStrip;
+import androidx.car.app.model.Distance;
+import androidx.car.app.model.Pane;
+import androidx.car.app.model.PaneTemplate;
+import androidx.car.app.model.Row;
+import androidx.car.app.model.Template;
+import androidx.car.app.navigation.model.Maneuver;
+import androidx.car.app.navigation.model.NavigationTemplate;
+import androidx.car.app.navigation.model.RoutingInfo;
+import androidx.car.app.navigation.model.Step;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.WritableMap;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.libraries.mapsplatform.turnbyturn.model.NavInfo;
+import com.google.android.libraries.mapsplatform.turnbyturn.model.StepInfo;
+import com.google.android.react.navsdk.AndroidAutoBaseScreen;
+import com.google.android.react.navsdk.NavInfoReceivingService;
+
+public class SampleAndroidAutoScreen extends AndroidAutoBaseScreen {
+  protected RoutingInfo mNavInfo;
+
+  public SampleAndroidAutoScreen(@NonNull CarContext carContext) {
+    super(carContext);
+
+    // Connect to the Turn-by-Turn Navigation service to receive navigation data.
+    NavInfoReceivingService.getNavInfoLiveData().observe(this, this::buildNavInfo);
+  }
+
+  /*
+   * This method is called when the Navigation data feed sends a new NavInfo object.
+   */
+  private void buildNavInfo(NavInfo navInfo) {
+    if (navInfo == null || navInfo.getCurrentStep() == null) {
+      return;
+    }
+
+    /**
+     * Converts data received from the Navigation data feed into Android-Auto compatible data
+     * structures. For more information see the "Ensure correct maneuver types" below.
+     */
+    Step currentStep = buildStepFromStepInfo(navInfo.getCurrentStep());
+    Distance distanceToStep =
+        Distance.create(max(navInfo.getDistanceToCurrentStepMeters(), 0), Distance.UNIT_METERS);
+
+    mNavInfo = new RoutingInfo.Builder().setCurrentStep(currentStep, distanceToStep).build();
+
+    // Invalidate the current template which leads to another onGetTemplate call.
+    invalidate();
+  }
+
+  @Override
+  public void onNavigationReady(boolean ready) {
+    super.onNavigationReady(ready);
+    // Invalidate template layout because of conditional rendering in the
+    // onGetTemplate method.
+    invalidate();
+  }
+
+  private Step buildStepFromStepInfo(StepInfo stepInfo) {
+    Maneuver.Builder maneuverBuilder = new Maneuver.Builder(stepInfo.getManeuver());
```

Comment:
It would be good to show how adding an icon works too. The bitmaps in StepInfo are compatible here so no need to convert and you can do something like: 

```
  IconCompat maneuverIcon =
               IconCompat.createWithBitmap(stepInfo.getManeuverBitmap());
  CarIcon maneuverCarIcon = new CarIcon.Builder(maneuverIcon).build();
  maneuverBuilder.setIcon(maneuverCarIcon);
```

Reply:
Done.